src/Common/src/CoreLib/System/IO/PathInternal.Windows.cs

```diff
@@ -401,7 +401,7 @@ internal static string NormalizeDirectorySeparators(string path)
         /// </summary>
         internal static bool IsEffectivelyEmpty(ReadOnlySpan<char> path)
         {
-            return (path.IsEmpty || path.IndexOf(' ') > -1);
+            return path.IsWhiteSpace();
```

Comment:
This is also a behavior change.  Why are you making this change?  How does this related to the title of this PR, which is about adding a test?

Reply:
The title could probably be more clear.  

I had to modify the mono fork to do this check:

https://github.com/mono/corefx/blob/9354dfd8d3cd4af62592bee9d8fe434e5694788a/src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs#L397-L402

I felt that it could be useful upstream to a.) add the test and b.) streamline IsEffectivelyEmpty a bit.

Reply:
But FSW isn't the only thing that's using this.  This change also impacts DirectoryInfo.CreateSubdirectory, Path.GetDirectoryName, Path.GetRelativePath, Path.GetFullPath, and so on.

cc: @JeremyKuhne

Reply:
Yes, correct.  Should I split that change into another PR?

Reply:
> I had to modify the mono fork to do this check

I don't see what about this test requires this change.  Can you elaborate?

Reply:
> Should I split that change into another PR?

No, but you should remove it from this one.  If you believe there's a bug in some public API, please open an issue to describe what that bug is.

Reply:
Yes, we definitely shouldn't change this. This code is attempting to prevent making API calls with the parent directory, which is what happens with an empty string. Windows eats all trailing spaces so it becomes "effectively" empty.

I'll make a PR that goes into detail.

Reply:
The driver for the change in the mono fork is that we started using the corefx fsw, but our Path validation is based on netfx rules and throws an ArgumentException.  I 'borrowed' the IsEffectivelyEmpty method and in the process saw that we could streamline.  In addition, I added a test to check the change and felt that both may be useful upstream.

If this doesn't have any value, I can pull the whole thing.

Reply:
We did make a number of changes to behavior to fix working with paths on Windows and to better work cross platform. I'd be careful about layering on altering behavior for Mono. There is a good chance Mono shouldn't or a chance we made some sort of mistake. :)